libs/components/molecules/src/lib/footer/footer.component.ts

```diff
@@ -0,0 +1,88 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { MatIconModule } from '@angular/material/icon';
+import { MatButtonModule } from '@angular/material/button';
+import { ConnectedPosition, OverlayModule } from '@angular/cdk/overlay';
+import { MatListModule } from '@angular/material/list';
+
+/**
+ * the interface has different formats for downloading which will be emitted once clicked on Download button
+
+ */
+export interface DownloadFormat {
+  /**
+   * format label type
+   */
+  label: string;
+}
+
+/**
+ * cdk overlay positioning to check the download list panel
+ * Added two panels for the footer component
+ */
+const DOWNLOADS_LIST_POSITION: ConnectedPosition[] = [
+  {
+    panelClass: 'above',
+    originX: 'start',
+    originY: 'top',
+    overlayX: 'start',
+    overlayY: 'bottom',
+    offsetY: 0,
+  },
+  {
+    panelClass: 'below',
+    originX: 'start',
+    originY: 'bottom',
+    overlayX: 'start',
+    overlayY: 'top',
+  },
+];
+
+/**
+ * Footer Component
```

Comment:
The documentation still needs to be improved. Here's what chatgpt has to say about this component:
<hr/>
This is a TypeScript code for a component called FooterComponent. It is a footer that appears at the bottom of a web page and contains several buttons and a download list panel.

The DownloadFormat interface is defined to specify the format label type.

The DOWNLOADS_LIST_POSITION constant is an array of ConnectedPosition objects. These objects specify the positioning of the download list panel relative to the button that triggers it.

The FooterComponent class is decorated with the @Component decorator, which provides metadata for the component. The component has the following properties:

downloadFormats is an input that allows the consumer to pass an array of DownloadFormat objects.
download is an output that emits a DownloadFormat object when the user clicks on an item in the download list panel.
illustrationClick, embedClick, contactClick, and hraPortalClick are outputs that emit events when the user clicks on the corresponding buttons.
DOWNLOADS_LIST_POSITION is a readonly property that initializes the download list position with the positions defined in DOWNLOADS_LIST_POSITION.
downloadListOpen is a boolean that determines whether the download list panel is open or closed.
The FooterComponent HTML template and styles are defined in separate files referenced by the templateUrl and styleUrls properties respectively.

Overall, this component is a reusable footer that allows the user to download files in different formats and provides buttons for navigating to different pages or triggering events.